RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.recentactivity;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import org.openide.modules.InstalledFileLocator;
+import org.openide.util.NbBundle;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
+import org.sleuthkit.autopsy.coreutils.ExecUtil;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.coreutils.PlatformUtil;
+import org.sleuthkit.autopsy.datamodel.ContentUtils;
+import org.sleuthkit.autopsy.ingest.DataSourceIngestModuleProcessTerminator;
+import org.sleuthkit.autopsy.ingest.IngestJobContext;
+import org.sleuthkit.autopsy.ingest.IngestServices;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+public class ExtractEdge extends Extract{
+    
+    private static final Logger logger = Logger.getLogger(ExtractIE.class.getName());
+    private final IngestServices services = IngestServices.getInstance();
+    private final String moduleTempResultsDir;
+    private Content dataSource;
+    private IngestJobContext context;
+    
+    private static String ESE_TOOL_NAME = "ESEDatabaseView.exe";
+    private static File ESE_TOOL_FILE;
+    private static String EDGE_WEBCACHE_NAME = "WebCacheV01.dat";
+    private static String EDGE = "Edge";
+
+    ExtractEdge() throws NoCurrentCaseException{
+        moduleName = NbBundle.getMessage(Chrome.class, "ExtractEdge.moduleName");
+        moduleTempResultsDir = RAImageIngestModule.getRATempPath(Case.getCurrentCaseThrows(), EDGE) + File.separator + "results";
+    }
+    
+    @Override
+    void process(Content dataSource, IngestJobContext context) {
+        this.dataSource = dataSource;
+        this.context = context;
+        dataFound = false;
+        
+        this.processWebCache();
+        
+         // Bookmarks come from spartan.edb different file
+        this.getBookmark(); // Not implemented yet
+    }
+    
+    void processWebCache(){
+        Path path = Paths.get("ESEDatabaseView", ESE_TOOL_NAME);
+        ESE_TOOL_FILE =  InstalledFileLocator.getDefault().locate(path.toString(), ExtractEdge.class.getPackage().getName(), false); //NON-NLS
+        if (ESE_TOOL_FILE == null) {
```

Comment:
I missed that string in path when I was cleaning up.  I have moved it to a constant and I will go through and add the NON-NLS as needed.

Yes, if the tool cannot be located we are done. 

I agree, I am reorganizing and writing some little functions.  I do too wish that process allowed throwing exceptions, it would make the code in process a little cleaner.